mava/components/jax/training/base.py

```diff
@@ -51,33 +51,3 @@ class Utility(Component):
     @abc.abstractmethod
     def on_training_utility_fns(self, trainer: SystemTrainer) -> None:
         """[summary]"""
-
-
-class Loss(Component):
-    @abc.abstractmethod
-    def on_training_loss_fns(self, trainer: SystemTrainer) -> None:
-        """[summary]"""
-
-    @staticmethod
-    def name() -> str:
-        """_summary_
-
-        Returns:
-            _description_
-        """
-        return "loss"
-
-
-class Step(Component):
-    @abc.abstractmethod
-    def on_training_step_fn(self, trainer: SystemTrainer) -> None:
-        """[summary]"""
-
-    @staticmethod
-    def name() -> str:
-        """_summary_
-
-        Returns:
-            _description_
-        """
-        return "sgd_step"
```

Comment:
Is there a reason we are moving the base components to 'step.py' and `losses.py`?

Reply:
Yup, I noticed that those particular base components were only used in those particular files, and didn't see why we should keep them entirely separate

Reply:
Ah I see, what are your thoughts on moving it back? Becuase once we have more systems they will probably all inheret from the same base `Step` and `Loss` components.

Reply:
It feels very weird to me to have `Step` and `Loss` not be in the files `step.py` and `losses.py` respectively. That's also the approach we take to all other abstract component classes.

Reply:
In that case, I agree with you. Please keep as is :)